codalab/model/bundle_model.py

```diff
@@ -2901,3 +2903,112 @@ def delete_oauth2_auth_code(self, auth_code_id):
             connection.execute(
                 oauth2_auth_code.delete().where(oauth2_auth_code.c.id == auth_code_id)
             )
+
+    # ===========================================================================
+    # Bundle Store methods follow!
+    # ===========================================================================
+    def get_bundle_stores(self, user):
+        with self.engine.begin() as connection:
+            rows = connection.execute(
+                select(
+                    [
+                        cl_bundle_store.c.uuid,
+                        cl_bundle_store.c.owner_id,
+                        cl_bundle_store.c.name,
+                        cl_bundle_store.c.storage_type,
+                        cl_bundle_store.c.storage_format,
+                        cl_bundle_store.c.url,
+                    ]
+                ).where(
+                    or_(
+                        cl_bundle_store.c.owner_id == 'codalab(0)',
+                        cl_bundle_store.c.owner_id == user,
+                    )
+                )
+            ).fetchall()
+            logger.error(rows)
+            return dict((row.uuid, (row.owner_id, row.name, row.url)) for row in rows)
+
+    def create_bundle_store(self, user, name, storage_type, storage_format, url, authentication):
+        """
+        create a bundle store
+        """
+        uuid = spec_util.generate_uuid()
+        with self.engine.begin() as connection:
+            bundle_store_value = {
+                'uuid': uuid,
+                'owner_id': user,
+                'name': name,
+                'storage_type': storage_type,
+                'storage_format': storage_format,
+                'url': url,
+                'authentication': authentication,
+            }
+            connection.execute(cl_bundle_store.insert().values(bundle_store_value))
+
+    def update_bundle_store(self, user, id, name, storage_type, storage_format, url, authentication):
+        """
+        update a bundle store
+        """
+        with self.engine.begin() as connection:
+            bundle_store_value = {
+                'owner_id': user,
+                'name': name,
+                'storage_type': storage_type,
+                'storage_format': storage_format,
+                'url': url,
+                'authentication': authentication,
+            }
+            connection.execute(
+                cl_bundle_store.update()
+                .where(cl_bundle_store.c.uuid == id)
```

Comment:
This is a problem because anyone can update any bundle store. We need to add a check to ensure that a user can only update a bundle store they own.